YapDatabase/YapDatabase.m

```diff
@@ -984,7 +989,30 @@ - (BOOL)configureEncryptionForDatabase:(sqlite3 *)sqlite
                 return NO;
             }
         }
-	}
+
+
+        if (options.legacyCipherCompatibilityVersion > 0) {
+            // `PRAGMA cipher_compatibility` is not available until SQLCipher 4.0.1 which corresponds
+            // to this version of sqlite.
+            // https://www.zetetic.net/blog/2018/12/18/sqlcipher-401-release/
+            const NSUInteger sqlCipherVersionSupportingCipherCompatibility = 3026000;
+            if (SQLITE_VERSION_NUMBER < sqlCipherVersionSupportingCipherCompatibility) {
+                YDBLogWarn(@"setting legacyCipherCompatibilityVersion on unsupported SQLCipher version has no effect.");
+            } else {
+                NSAssert(!hasCustomCipherConfiguration, @"If you are specifying custom cipher parameters, you cannot use legacyCipherCompatibility. Instead you must specify *all* relevant cipher params. Details: https://discuss.zetetic.net/t/upgrading-to-sqlcipher-4/3283");
```

Comment:
This hunk is the substantial change, and here's where we assert to try to help users not misuse the feature.